h/templates/accounts/signup-post.html.jinja2

```diff
@@ -49,7 +49,10 @@
 {% block scripts %}
   {{ super() }}
 
-  {% for url in asset_urls('login_forms_js') %}
+  {# Client-rendered frontend is only used if signup was not successful. #}
+  {% if js_config.get("form") %}
+    {% for url in asset_urls('login_forms_js') %}
     <script type="module" src="{{ url }}"></script>
-  {% endfor %}
+    {% endfor %}
+  {% endif %}
```

Comment:
A possible future change is to render the frontend using JS for both successful and unsuccessful sign-up form submissions, in which case this check can be removed.